Library/Homebrew/extend/os/linux/extend/ENV/super.rb

```diff
@@ -20,4 +20,25 @@ def determine_rpath_paths
     paths += homebrew_extra_library_paths
     paths.to_path_s
   end
+
+  def x11
+    xorg = Formula["linuxbrew/xorg/xorg"]
+    if xorg.installed?
+      xorg.recursive_dependencies.each do |d|
+        dep_formula = d.to_formula
+        append_path "PKG_CONFIG_LIBDIR", dep_formula.lib/"pkgconfig"
+        append_path "PKG_CONFIG_LIBDIR", dep_formula.share/"pkgconfig"
```

Comment:
These are borrowed from Mac's X11 stdenv. The basic gist of it is you need to add every recursive dependency of `xorg` to the various paths.

Note that it *needs* to say `recursive_dependencies` and not just `runtime_dependencies`; otherwise `libxcb` doesn't add `libxmcp` to the list and thus pkg-config blows up for it.

The relevant command from the configure script is `/home/linuxbrew/.linuxbrew/opt/pkg-config/bin/pkg-config --exists --print-errors 'xcb >= 1.6 xcb-render >= 1.6'`